pkg/api/defaults.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+func init() {
+	Scheme.AddDefaultingFuncs(
+		func(obj *Volume) {
+			if util.AllPtrFieldsNil(&obj.Source) {
+				obj.Source = VolumeSource{
+					EmptyDir: &EmptyDir{},
+				}
+			}
+		},
+		func(obj *Port) {
+			if obj.Protocol == "" {
+				obj.Protocol = ProtocolTCP
+			}
+		},
+		func(obj *Container) {
+			if obj.ImagePullPolicy == "" {
+				// TODO(dchen1107): Move ParseImageName code to pkg/util
+				parts := strings.Split(obj.Image, ":")
+				// Check image tag
+				if parts[len(parts)-1] == "latest" {
+					obj.ImagePullPolicy = PullAlways
+				} else {
+					obj.ImagePullPolicy = PullIfNotPresent
+				}
+			}
+			if obj.TerminationMessagePath == "" {
+				obj.TerminationMessagePath = TerminationMessagePathDefault
+			}
+		},
+		func(obj *RestartPolicy) {
+			if util.AllPtrFieldsNil(obj) {
+				obj.Always = &RestartPolicyAlways{}
+			}
+		},
```

Comment:
DNSPolicy default should be set here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/v1beta1/types.go#L65
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/validation/validation.go#L378